verification/policy_file_processor.go

```diff
@@ -0,0 +1,338 @@
+
+	for _, name := range names {
+		if _, ok := existingPolicies[name]; !ok {
+			return nil, ErrorPolicyNotExists{
+				Msg: fmt.Sprintf("%s not exists.", name),
+			}
+		}
+		policies = append(policies, existingPolicies[name])
+	}
+
+	return policies, nil
+}
+
+// LoadDefaultPolicyDocument creates a PolicyDocument from default TrustPolicy
+// file.
+func LoadDefaultPolicyDocument() (*PolicyDocument, error) {
+	policyDocument := &PolicyDocument{}
+	jsonFile, err := os.Open(trustPolicyPath)
+	if err != nil {
+		return nil, err
+	}
+	defer jsonFile.Close()
+	err = json.NewDecoder(jsonFile).Decode(policyDocument)
+	if err != nil {
+		return nil, err
+	}
+	return policyDocument, nil
+}
+
+// save stores the trust policy to file.
+// TODO: move to config submodule.
+func (c *PolicyDocument) save() error {
+	dir := filepath.Dir(trustPolicyPathForWrite)
+	if err := os.MkdirAll(dir, 0700); err != nil {
+		return err
+	}
+	file, err := os.Create(trustPolicyPathForWrite)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+	encoder := json.NewEncoder(file)
+	encoder.SetIndent("", "    ")
+	return encoder.Encode(c)
+}
+
+func (pd *PolicyDocument) getNameToPolicyMap() map[string]*TrustPolicy {
+	policies := make(map[string]*TrustPolicy)
+	for _, policy := range pd.TrustPolicies {
+		newPolicy := policy
+		policies[policy.Name] = &newPolicy
+	}
+	return policies
+}
+
+// overwritePolicies replaces TrustPolicies of PolicyDocument with the given
+// policies.
+func (pd *PolicyDocument) overwritePolicies(newPolicies map[string]*TrustPolicy) {
+	policies := make([]TrustPolicy, 0, len(newPolicies))
+	for _, policy := range newPolicies {
+		policies = append(policies, *policy)
+	}
+	pd.TrustPolicies = policies
+}
+
+// NewTrustPolicy creates a new TrustPolicy with speicified fields.
+func NewTrustPolicy(name string, scopes []string, level string, override map[string]string, stores []string, identities []string) *TrustPolicy {
+	return &TrustPolicy{
+		Name:           name,
+		RegistryScopes: scopes,
+		SignatureVerification: SignatureVerification{
+			Level:    level,
+			Override: override,
+		},
+		TrustStores:       stores,
+		TrustedIdentities: identities,
+	}
+}
+
+// LoadTrustPolicies loads policies from the given file.
+func LoadTrustPolicies(policyPath string) ([]*TrustPolicy, error) {
+	policyBytes, err := os.ReadFile(policyPath)
+	if err != nil {
+		return nil, err
+	}
+	policyBytes = bytes.Trim(policyBytes, " ")
+
+	version, err := extractPolicyVersion(policyBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return ParseTrustPolicies(policyBytes, version)
+}
+
+// extractPolicyVersion extracts version by parsing the policy bytes.
+func extractPolicyVersion(policyBytes []byte) (string, error) {
+	var policy map[string]interface{}
+	if err := json.Unmarshal(policyBytes, &policy); err != nil {
+		return "", err
+	}
+	version, ok := policy["version"]
+	if !ok {
+		return "", errors.New("version is not specified")
+	}
+	return version.(string), nil
```

Comment:
Can we unmarshal it to the `PolicyDocument` instead of a map?

Reply:
This is for compatibility. The `policyBytes` can be in different schemas, so we cannot just unmarshal it to a specific struct.

Reply:
> This is for compatibility. The `policyBytes` can be in different schemas, so we cannot just unmarshal it to a specific struct.

Do we need to check the type of `version` here?

Reply:
I don't have a strong opinion on it. I was thinking to check boolean value returned by type assertion, but felt it's a bit over checked. Let's wait for more comments on it.